apps/sms/views/shared/js/shim_host_bootstrap.js

```diff
@@ -7,6 +8,12 @@
 'use strict';
 
 exports.bootstrap = function(appInstanceId) {
+  // We can't handle system messages inside iframe right now, so we access
+  // parent window here. This should be removed once bug 818000 is resolved.
+  ActivityShim.init(
```

Comment:
IIRC that's the main reason why we still leave ActivityShim in window. So back to the original problem about the memory consumption: Will using the optimized client in window could reduce further than optimized bridge in bug 1218091? If the effect is limited, maybe we can still land bug 1218091 first and keep this one with using only client in the window after bug 818000, wdyt?